client/allocrunner/alloc_runner.go

```diff
@@ -482,6 +487,8 @@ func (ar *allocRunner) handleTaskStateUpdates() {
 			}
 		}
 
+		ar.taskHookCoordinator.taskStateUpdated(states)
```

Comment:
We should compute states and call `taskStateUpdated(states)` in `Restore()` method as well to handle restore and client restarts as well.